src/OSPSuite.UI/Services/DataImporter.cs

```diff
@@ -169,5 +177,45 @@ DataImporterSettings dataImporterSettings
             return result;
          }
       }
+
+      public Cache<string, IEnumerable<DataRepository>> ReloadFromConfiguration(IEnumerable<DataRepository> dataSetsToImport,
+         IEnumerable<DataRepository> existingDataSets)
+      {
+         var result = new Cache<string, IEnumerable<DataRepository>>();
+
+         var newDataSets = dataSetsToImport.Where(dataSet => !repositoryExistsInList(existingDataSets, dataSet));
+         result.Add(Constants.ImporterConstants.DATA_SETS_TO_BE_NEWLY_IMPORTED, newDataSets);
+
+         var dataSetsToBeDeleted = existingDataSets.Where(dataSet => !repositoryExistsInList(dataSetsToImport, dataSet));
+         result.Add(Constants.ImporterConstants.DATA_SETS_TO_BE_DELETED, dataSetsToBeDeleted);
+
+         var overwrittenDataSets = dataSetsToImport.Except(newDataSets);
+         result.Add(Constants.ImporterConstants.DATA_SETS_TO_BE_OVERWRITTEN, overwrittenDataSets);
+
+         using (var reloadPresenter = _applicationController.Start<IImporterReloadPresenter>())
+         {
+            reloadPresenter.AddDeletedDataSets(dataSetsToBeDeleted.AllNames());
+            reloadPresenter.AddNewDataSets(newDataSets.AllNames());
+            reloadPresenter.AddOverwrittenDataSets(overwrittenDataSets.AllNames());
+            reloadPresenter.Show();
+
+            if (reloadPresenter.Canceled())
+               return new Cache<string, IEnumerable<DataRepository>>()
+               {
+                  {Constants.ImporterConstants.DATA_SETS_TO_BE_DELETED, Enumerable.Empty<DataRepository>()}, 
+                  {Constants.ImporterConstants.DATA_SETS_TO_BE_NEWLY_IMPORTED,Enumerable.Empty<DataRepository>()},
+                  {Constants.ImporterConstants.DATA_SETS_TO_BE_OVERWRITTEN,Enumerable.Empty<DataRepository>()},
+               };
+         }
+
+         return result;
+      }
+
+
+      private bool repositoryExistsInList(IEnumerable<DataRepository> dataRepositoryList, DataRepository targetDataRepository)
+      {
+         return dataRepositoryList.Select(dataRepo => targetDataRepository.ExtendedProperties.KeyValues.All(keyValuePair => dataRepo.ExtendedProperties[keyValuePair.Key].ValueAsObject.ToString() == keyValuePair.Value.ValueAsObject.ToString())).Any(result => result);
```

Comment:
Why the ToString() in the comparison
Why the any(x=>x). I don't understand the construct x=>x
This can be rewritten with dataRepositoryLiust.Any(x=>xxxx)

Reply:
corrected.